sql/sql_connect.cc

```diff
@@ -346,6 +312,60 @@ void init_global_index_stats(void)
   }
 }
 
+static void reset_global_user_or_client_stats(HASH* stats)
+{
+  DYNAMIC_ARRAY delete_items;
+
+  if (unlikely(my_init_dynamic_array(&delete_items, sizeof(USER_STATS *),
+        stats->records, 1)))
+  {
+    sql_print_error("reset global stats: cannot initialize dynamic array");
+    my_message(ER_OUTOFMEMORY, ER(ER_OUTOFMEMORY), MYF(0));
+    return;
+  }
+
+  for (ulong idx=0; idx < stats->records; idx++)
+  {
+    USER_STATS* user_stats=
+      reinterpret_cast<USER_STATS*>(my_hash_element(stats, idx));
+    if (user_stats->concurrent_connections)
+    {
+      uint saved_concurrent_connections = user_stats->concurrent_connections;
+      init_user_stats(user_stats);
```

Comment:
The need for saved_concurrent_connections would disappear with init_user_stats taking a concurrent_connections arg